.travis.yml

```diff
@@ -8,12 +8,30 @@ env:
+  - secure: O7KZwRH1bowB2tdv6GPzy6Ge/u8xE+mL2wPhkCpcHimXfH21gO0T+sIoM4RiomrD/WNQKKdeZsX43W6ZKwx1Dsb3OoLnsKF8biMiAftiJ+ROryy+/m2yy08RY9wOtBAYrYij3lvWNCbOif0unZ0d82sqhcrs0LN/GGkCv+T65PBKCn65SHIi4EboS9Oce4ZPVq6cyo8wliKLRFrSElgBDoVKyrjSPLu84bP9SjPVO5Ycq9QJ/ts/Ns6H/G+u/k9GmsnBQAQafxjSYNmZduVhToRb7n+BDgJb6B9M+ujXQwZgMjQBSVlRdnHdLEXh6EE2RhC6icizwd4MPA1cZUOSDhOXm0RzvekOiO83E37sn4Yv2tsEIDGHqag8y6QiwjYHlzxP5TQCJauo54iyzN8JEwEAp6tGh8phiKnaMJiMCvmeo2uz/f80k+sUh4urxKTSFIA+fD/9g09SujqqAgxyGQ9Ao6yxxctjqlM9equXu29iWNw/aje8ygCxWyxE/u87SCS8xzeZNKhadYmLduGjDcT2g7k2kcI75aOFJ/bMrIcDm4YKc83JUghRHw+MkR5YdD7zdNvnYG0NIxu1c9Ajv657X25hP8OBt+IcJPVmUgTJRLbGIvcsYqQZY1J7LPM6r6KgXmQ8X2G7CfARix4WVTiUVH+em6ZxviFoy039XQo=
+  - secure: SWTJB/ZdAmn9v1oaWHUEVYMZ3ImkFRlqHibnigT6wwxs4np+BvObfBxJ+FOMTFasoSqO/W2OKdrrf5r3cKmmqjPgF/g7x4FqYZ5VJsQIZcZlvTgaHkyZCeBlNpfm/JeWUAeG7QB5tOwKofjo0ws2OYmFCa2cgO0GBRs/ZPH6/z9PaPoLAQ5x+jETDo3bOCdz8iB4BbJajuznbHrloa9RezhwEWDk8eIr7Izl9wXT0VUGDkQIUVkv/uvFHScXtn9RLhqzLWHCiwLWZJf6N1pIwuYhiYpjypXuhqQ8g58b2gLAQbbIUg+0M1PT4rRx6veXPdXYd2ZtA0p/roR+9/HmT/jsf7qwaMCthVUn/CKL6X1sFDSRWPpT590EAPnw/iqOjxiqHFRcsnujjBLHhegFy/+LuqjIKBg7S9ORtCgJifoKDlvGzifcdiXC31aENQOmAbRYav5f4CuCuV2rHiuCmLpIYmPj9FBllSGqE6J2qqTnng+9ZaF3KvCj3XlVKbRqjv/6sgApuBS62z4mljWJAmO6mcJ4QU1LApg7UFXeY7goVwdWAhIGkqjs4LZy6ZkmaOw3q1n50TdFBpDa8IatSocFbT+i7MG/vDBTOHIK26w2mAtFzHALrACgy57nHEhSIejmOlSeJgXwox75PLqP2xGw/lEyeJatD8kkKj+6zMg=
+  - secure: jmBwyPpZlIweGsacx4I5npe4SjO+emUkJE9RWuWHmQPT6QlRggMSBFgLIR7CeznLvPkjgVWvVAQV0nsQfS4LIsknIloC4Ew81GNCjll7PR8KHzTFZpvP/ilY+ZXt8NG/jHHJdrMao1aeFHkTEOEhTruccWdDSpa6JHV7emdf3Do6JaKGPxaatik+k5DGfUBdiLhU8t2EW3SK7oEtMo2YZ6FGHa+v7bohPenE1pJdHhGurWcndkLpz+TsOD5++dgrBQS2qaioXt4derFzwmAs7R1tNSsWxU+jvkoyXUZNdsa2K1lrzcOAgIqIpztOr3QtJ4F/RylEOSTYlLQSSHchPYkfRI2CRK7JsZE1KTOoVHta6w6l/MvG4glaRBS61adj/p7R7iurBPjWW1uJOx5ti5tL2y4x9LpIlaE/PprhWtM5f2jXpkitqUX4t1puuedkp9a1tQD7GxSqr1XrBy9qmuUZTS2VsjGVDtVdtjiqziq1/ysLFtV9hp6jKHja1kb5N17f6zucoIHSn4Gp/SlkTSe0/DkJAjp4NVUVB5mm/AyT535cQMdmM8GuN1J66KFMCtMfhnE8BaLmSNwLK99V/57yvWAAERGjeeKOxCh+hfmgF5+9JB0AyiBoUhxmxQw1hRn4fxrVwrRM3jMkiEOUR6By1bECO2xTvZEgPqfrOus=
 install: true
 before_script:
 - gpg --import arpnetworking.key
 script:
 - curl -s ${JDKW} | bash /dev/stdin ./mvnw clean install --settings settings.xml -P ci -Dgpg.skip=true -U
-- if [ \( -n "${TRAVIS_TAG}" \) -a "${TRAVIS_PULL_REQUEST}" = "false" ]; then curl -s ${JDKW} | bash /dev/stdin ./mvnw deploy --settings settings.xml -P ci -DskipAllVerification=true; fi;
+- if [ \( -n "${TRAVIS_TAG}" \) -a "${TRAVIS_PULL_REQUEST}" = "false" ]; then
+-   curl -s ${JDKW} | bash /dev/stdin ./mvnw deploy --settings settings.xml -P ci -DskipAllVerification=true
+-   if [ $? -eq 0]; then
```

Comment:
Not sure this is going to work nicely with the pipe above. I recall we had issues with this before. Even if you don't want to put all these commands in pom.xml I think they should be externalized and kicked-off from the pom (e.g. via profile) to avoid exactly this type of issue. For example:

```bash
TARGET="install"
PROFILES="ci,"
if [ \( -n "${TRAVIS_TAG}" \) -a "${TRAVIS_PULL_REQUEST}" = "false" ]; then
    TARGET="${TARGET}docker,"
    PROFILES="${PROFILES}
fi
curl -s ${JDKW} | bash /dev/stdin ./mvnw clean ${TARGET} --settings settings.xml -P ${PROFILES} -Dgpg.skip=true -U
```

Then the docker profile adds the creation and publication of a docker image. This can be a script execution for now, but it's at least insulated behind the same standard entry point if I want to build this locally instead of tee'd off from the travis specific entry point.

What's wrong with this one?

https://github.com/spotify/docker-maven-plugin

- Supports docker files
- Supports multiple tags

Looks like we should use an assembly to build the layout (https://maven.apache.org/plugins/maven-assembly-plugin/directory-mojo.html) and then either add it directly to the image or add from it. This combination would allow us to nicely layer base image, dependencies, config, and code separately in the right order.